src/components/Extension/CloudinaryExtension/components/CloudinaryUploadWidget.tsx

```diff
@@ -0,0 +1,45 @@
+import { Button } from '@awell_health/ui-library'
+import { useMemo } from 'react'
+import { useScript } from '../../../../hooks/useScript'
+import { openUploadWidget } from '../utils/CloudinaryService'
+import { UploadWidgetOptions } from '../utils/types'
+
+interface CloudinaryUploadWidgetProps extends UploadWidgetOptions {
+  onFileUpload: (publicId: string) => void
+}
+
+export const CloudinaryUploadWidget = ({
+  onFileUpload: onFileUpload,
+  ...widgetOptions
+}: CloudinaryUploadWidgetProps) => {
+  const { isLoaded } = useScript(
+    'https://widget.cloudinary.com/v2.0/global/all.js'
+  )
+
+  const widget = useMemo(() => {
+    return isLoaded
+      ? openUploadWidget(
+          {
+            ...widgetOptions,
+            sources: ['local'],
+          },
+          function (error: unknown, result: Record<string, unknown>) {
+            if (!error && result?.event === 'success') {
+              //@ts-expect-error
```

Comment:
Add an explanation why `//@ts-expect-error - some explanation` is needed.

Reply:
Actually that change was already there when I started working on it. I changed that function a little bit, but I built it on top of your PoC. So I can't give you an explanation why is it needed. If you want, I can remove it.

Reply:
If it can be removed, then let's remove it idd.